### PR TITLE
Update deps and structure

### PR DESCRIPTION
**Pull request by BastianZim**
Checklist
* [x] Used a [personal fork of the feedstock to propose changes](https://conda-forge.org/docs/maintainer/updating_pkgs.html#forking-and-pull-requests)
* [x] Bumped the build number (if the version is unchanged)
* [ ] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

### CODE REVIEW

**Comment by BastianZim** [comment]
@conda-forge-admin, please rerender

**Comment by BastianZim** [comment]
Created with Grayskull

File: recipe/meta.yaml [hunk 1]
```diff
@@ -11,45 +12,48 @@ source:
...
+    - importlib-metadata
     - multiprocess
     - numpy >=1.17
-    - openpyxl
     - packaging
     - pandas
-    - py7zr
-    - pyarrow >=0.17.1
+    - pyarrow >=1.0.0,!=4.0.0
+    - python
```
[the remark this answers is not in the document]
Comment (BastianZim) [reply recipe/meta.yaml]: Oops missed that, thanks.

**Comment by BastianZim** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 2]
```diff
@@ -11,45 +12,47 @@ source:
...
+    - pip
 
 about:
   home: https://github.com/huggingface/datasets
   license: Apache-2.0
   license_family: Apache
-  # License file manually packaged. See: https://github.com/huggingface/datasets/pull/1007
   license_file: LICENSE
   summary: HuggingFace/Datasets is an open library of NLP datasets.
   doc_url: https://huggingface.co/docs/datasets/
```
Comment (thewchan) [inline recipe/meta.yaml:58]: Lets add a `description` entry here, just an copy-and-paste (slightly edited) from upstream:
> Datasets is a lightweight library providing one-line dataloaders for many public datasets and one liners to download and pre-process any of the number of datasets major public datasets provided on the HuggingFace Datasets Hub. Datasets are ready to use in a dataloader for training/evaluating a ML model (Numpy/Pandas/PyTorch/TensorFlow/JAX). Datasets also provide an API for simple, fast, and reproducible data pre-processing for the above public datasets as well as your own local datasets in CSV/JSON/text.

**Review by thewchan — CHANGES_REQUESTED** [review]

**Comment by BastianZim** [comment]
@conda-forge-admin, please rerender

**Comment by thewchan** [comment]
Thanks @BastianZim

**Pull request merged**